crates/matrix-sdk-crypto/src/store/caches.rs

```diff
@@ -197,8 +198,9 @@ impl DeviceStore {
 /// subtraction. For example, suppose we've just overflowed from i64::MAX to
 /// i64::MIN. (i64::MAX.wrapping_sub(i64::MIN)) is -1, which tells us that
 /// i64::MAX comes before i64::MIN in the sequence.
-#[derive(Clone, Copy, Debug, Default, PartialEq, Eq)]
-pub(crate) struct SequenceNumber(i64);
+#[derive(Clone, Copy, Debug, Default, PartialEq, Eq, Deserialize, Serialize)]
+#[serde(transparent)]
+pub struct SequenceNumber(i64);
```

Comment:
I had to make `SeuenceNumber` public, since it's used by other things that are public (https://github.com/matrix-org/matrix-rust-sdk/actions/runs/9685106739/job/26724426654)  Though I have a suspicion that those things aren't used outside of this crate, so maybe it's better to make those things `pub(crate)` instead?

Reply:
I think they are pub because store implementations might use them to serialize things more efficiently. I'm sure there are things that are needlessly public but this might not be such a situation.